dev/AppNotifications/AppNotificationManager.cpp

```diff
@@ -95,6 +96,18 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
         try
         {
+            {
+                auto lock{ m_lock.lock_exclusive() };
+                THROW_HR_IF_MSG(HRESULT_FROM_WIN32(ERROR_OPERATION_IN_PROGRESS), m_registering, "Registration is in progress!");
+                m_registering = true;
+            }
+
+            auto registeringScopeExit{ wil::scope_exit([&]()
+            {
+                auto lock { m_lock.lock_exclusive() };
+                m_registering = false;
```

Comment:
Should we be worried about unregistering on failures?

If a notificationPlatform call fails, registry will still have all the entries (I believe we have a failsafe for this). If AddToastRegistrationMapping succeeds and we throw later, the entry will be in the Long Running Process. I'm not super worried about this since the Long Running Process has its own cleanup functions, but how bad is it really? 
And we have called PushNotification_RegisterFullTrustApplication. Do we want to unregister if anything in Register() fails?

Something to think about.

Reply:
Yeah i think we can go all the way for cleanup. But at the end of the day, all this is a best effort. If Register fails, there is a good chance that Unregister fails as well (that typically happens if the plaform is in a bad state or has crashed.). So I don't believe we have a perfect solution.

Reply:
Talked offline. We will file a bug to do best effort cleanup similar to PushManager::Register @pmpurifoy

Reply:
https://github.com/microsoft/WindowsAppSDK/issues/2269